crates/typst/src/math/underover.rs

```diff
@@ -267,29 +403,29 @@ fn layout_underoverspreader(
     let stretched = glyph.stretch_horizontal(ctx, body.width(), Abs::zero());
 
     let mut rows = vec![MathRun::new(vec![body]), stretched.into()];
+    if let Some(annotation) = annotation {
+        let (above_style, below_style);
+        let annot_styles = match position {
+            Position::Above => {
+                above_style = style_for_subscript(styles);
```

Comment:
If the annotation is above, it should use `superscript` style, if it is below, it should use subscript style.